packages/lit-dev-content/src/components/ts-js.ts

```diff
@@ -0,0 +1,58 @@
+import {html, LitElement} from 'lit';
+import {customElement, state} from 'lit/decorators.js';
+import {
+  getCodeLanguagePreference,
+  CODE_LANGUAGE_CHANGE,
+} from '../code-language-preference.js';
+
+/**
+ * A custom element that will conditonally display slotted content based on the
+ * current code language preference. Defaults to displaying `js` and `ts`.
+ *
+ * @example
+ * ```html
+ * <article>
+ *   The file that deals with the code's metadata is:
+ *   index.<ts-js
+ *     ><span slot="js">min.js</span
+ *     ><span slot="ts">d.ts</span
+ *   ></ts-js>
+ *   <!-- Do not have whitespace to prevent awkward whitespace between "index."
+ *   and the displayed output of ts-js !-->
+ * ```
+ */
+@customElement('ts-js')
+export class LitDevCodeLanguageDisplay extends LitElement {
```

Comment:
Should the class name be updated? I don't know if `TsJs` is a great class name though.

Reply:
I think TsJsLanguageDisplay is good?